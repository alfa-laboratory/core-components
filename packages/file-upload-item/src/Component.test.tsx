import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { FileUploadItem, FileStatuses } from './index';

export const fileProps = {
    name: 'Довольно длинное название файла.pdf',
    uploadDate: '22.01.2018',
    size: 45000,
    downloadLink: '/link',
    uploadStatus: 'SUCCESS' as FileStatuses,
    showDelete: true,
};

describe('FileUploadItem', () => {
    describe('Snapshots tests', () => {
        it('should match snapshot', () => {
            expect(render(<FileUploadItem {...fileProps} />)).toMatchSnapshot();
        });
    });

    it('should set `data-test-id` attribute', () => {
        const dataTestId = 'test-id';
        const { queryByTestId } = render(<FileUploadItem dataTestId={dataTestId} />);

        expect(queryByTestId(dataTestId)).toBeInTheDocument();
    });

    it('should set `className` class', () => {
        const className = 'test-class';
        const { container } = render(<FileUploadItem className={className} />);

        expect(container.firstElementChild).toHaveClass(className);
    });

    it('should set `className` class', () => {
        const className = 'test-class';
        const { container } = render(<FileUploadItem className={className} />);

        expect(container.firstElementChild).toHaveClass(className);
    });

    describe('Callbacks tests', () => {
        it('should call `onDelete` prop', () => {
            const cb = jest.fn();
            const fileId = 'id';
            const { getByLabelText } = render(
                <FileUploadItem showDelete={true} onDelete={cb} id={fileId} />,
            );

            fireEvent.click(getByLabelText('удалить'));

            expect(cb).toBeCalledTimes(1);
            expect(cb.mock.calls[0][0]).toBe(fileId);
        });

        it('should call `onRestore` prop', () => {
            const cb = jest.fn();
            const fileId = 'id';
            const { getByText } = render(
                <FileUploadItem showRestore={true} onRestore={cb} id={fileId} />,
            );

            fireEvent.click(getByText('Восстановить'));

            expect(cb).toBeCalledTimes(1);
            expect(cb.mock.calls[0][0]).toBe(fileId);
        });

        it('should call `onRestore` prop', () => {
            const cb = jest.fn();
            const fileId = 'id';
            const { getByText } = render(
                <FileUploadItem {...fileProps} downloadLink='/link' onDownload={cb} id={fileId} />,
            );

            fireEvent.click(getByText(fileProps.name));

            expect(cb).toBeCalledTimes(1);
            expect(cb.mock.calls[0][0]).toBe(fileId);
        });
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<FileUploadItem />);

        expect(unmount).not.toThrowError();
    });
});
