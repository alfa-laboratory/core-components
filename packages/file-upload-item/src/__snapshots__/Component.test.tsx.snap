// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploadItem Snapshots tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component success"
      >
        <div
          class="info"
        >
          <svg
            class="successIcon"
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              clip-rule="evenodd"
              d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zm-1.5-6L17 9.5 15.5 8l-5 5-2-2L7 12.5l3.5 3.5z"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="name"
          >
            <a
              class="component primary pseudo"
              href="/link"
            >
              <span
                class="text"
              >
                Довольно длинное название файла.pdf
              </span>
            </a>
          </div>
          <div
            class="meta"
          >
            <span>
              22.01.2018
            </span>
            <span
              class="size"
            >
              43.95 KB
            </span>
          </div>
        </div>
        <button
          aria-label="удалить"
          class="component ghost s iconOnly delete primary"
          type="button"
        >
          <span
            class="addons"
          >
            <span
              class="iconWrapper xxs"
            >
              <svg
                class="icon"
                fill="currentColor"
                focusable="false"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M10.586 12L6 16.5 7.5 18l4.5-4.586L16.5 18l1.5-1.5-4.586-4.5L18 7.5 16.5 6 12 10.586 7.5 6 6 7.5l4.586 4.5z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component success"
    >
      <div
        class="info"
      >
        <svg
          class="successIcon"
          fill="currentColor"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zm-1.5-6L17 9.5 15.5 8l-5 5-2-2L7 12.5l3.5 3.5z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="name"
        >
          <a
            class="component primary pseudo"
            href="/link"
          >
            <span
              class="text"
            >
              Довольно длинное название файла.pdf
            </span>
          </a>
        </div>
        <div
          class="meta"
        >
          <span>
            22.01.2018
          </span>
          <span
            class="size"
          >
            43.95 KB
          </span>
        </div>
      </div>
      <button
        aria-label="удалить"
        class="component ghost s iconOnly delete primary"
        type="button"
      >
        <span
          class="addons"
        >
          <span
            class="iconWrapper xxs"
          >
            <svg
              class="icon"
              fill="currentColor"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M10.586 12L6 16.5 7.5 18l4.5-4.586L16.5 18l1.5-1.5-4.586-4.5L18 7.5 16.5 6 12 10.586 7.5 6 6 7.5l4.586 4.5z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
